From 39927132729bcd8569716f4ccd3a2e3eadfaf186 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Mon, 8 May 2017 15:54:46 +0200
Subject: Map evdev keycodes KEY_RFKILL and KEY_WWAN to XF86RFKill and XF86WWAN

In particular KEY_RFKILL is required for properly handling airplane mode
in userspace.

https://bugs.freedesktop.org/show_bug.cgi?id=100970
---
 keycodes/evdev | 2 ++
 symbols/inet   | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/keycodes/evdev b/keycodes/evdev
index 624ac68..51a6e26 100644
--- a/keycodes/evdev
+++ b/keycodes/evdev
@@ -281,6 +281,8 @@ default xkb_keycodes "evdev" {
 	<I251> = 251;	// #define KEY_BRIGHTNESS_CYCLE    243
 	<I252> = 252;	// #define KEY_BRIGHTNESS_ZERO     244
 	<I253> = 253;	// #define KEY_DISPLAY_OFF         245
+	<I254> = 254;	// #define KEY_WWAN                246
+	<I255> = 255;	// #define KEY_RFKILL              247
 
 	// Fake keycodes for virtual keys
 	<LVL3> =   92;
diff --git a/symbols/inet b/symbols/inet
index 5403e77..2934b65 100644
--- a/symbols/inet
+++ b/symbols/inet
@@ -215,8 +215,8 @@ xkb_symbols "evdev" {
 //  key <I251>   {      [ ]       }; // KEY_BRIGHTNESS_CYCLE  --  bright up, max++ == min
 //  key <I252>   {      [ ]       }; // KEY_BRIGHTNESS_ZERO  --  brightness off
 //  key <I253>   {      [ ]       }; // KEY_DISPLAY_OFF  --  turn off display
-//  key <I254>   {      [ ]       }; // KEY_WIMAX
-    key <I255>   {      [ XF86Hibernate         ]       }; // KEY_HIBERNATE
+    key <I254>   {      [ XF86WWAN              ]       }; // KEY_WWAN
+    key <I255>   {      [ XF86RFKill            ]       }; // KEY_RFKILL
 
     key <FK13>   {      [ XF86Tools         ]       };
     key <FK14>   {      [ XF86Launch5       ]       };
-- 
cgit v1.1


